src/mpid/ch4/generic/am/mpidig_am_recv_utils.h

```diff
@@ -274,6 +288,18 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_recv_copy_seg(void *payload, MPI_Aint payloa
     }
 }
 
+MPL_STATIC_INLINE_PREFIX bool MPIDIG_recv_initialized(MPIR_Request * rreq)
+{
+    return MPIDIG_REQUEST(rreq, req->recv_async).recv_type != MPIDIG_RECV_NONE;
+}
+
+MPL_STATIC_INLINE_PREFIX void MPIDIG_recv_set_data_copy_cb(MPIR_Request * rreq,
+                                                           MPIDIG_recv_data_copy_cb cb)
+{
+    MPIDIG_REQUEST(rreq, req->recv_async).data_copy_cb = cb;
+    MPIDIG_REQUEST(rreq, req->status) &= ~MPIDIG_REQ_BUSY;
```

Comment:
Let me think how to avoid this leaking. The busy flag was to prevent another thread to match an unexpected request that has not finished its ingress processing in the workq model.

Reply:
never mind. fixed it.